### PR TITLE
Fix Nestmates to throw correct exception type

### PR DESCRIPTION
**Pull request by fengxue-IS**
- Check for Nesthost exception before throwing illegalAccessException

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

File: runtime/vm/resolvesupport.cpp [hunk 1]
```diff
@@ -352,17 +352,24 @@ resolveClassRef(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cpIndex, UDAT
...
-			if (canRunJavaCode) {
-				char *errorMsg = NULL;
-				PORT_ACCESS_FROM_VMC(vmStruct);
-				if (J9_VISIBILITY_NON_MODULE_ACCESS_ERROR == checkResult) {
-					errorMsg = illegalAccessMessage(vmStruct, accessModifiers, J9_CLASS_FROM_CP(ramCP), accessClass, J9_VISIBILITY_NON_MODULE_ACCESS_ERROR);
-				} else {
-					errorMsg = illegalAccessMessage(vmStruct, -1, J9_CLASS_FROM_CP(ramCP), accessClass, checkResult);
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+			/* Check for pending exception for Nesthost class loading / verify, do not overwrite these exceptions */
+			if (!VM_VMHelpers::exceptionPending(vmStruct)) {
```
Comment (DanHeidinga) [inline runtime/vm/resolvesupport.cpp]: I think we want to change the order of these checks - `canRunJavaCode` should be first as their had better not be a pending exception if we can't run java code.

Comment (DanHeidinga) [reply runtime/vm/resolvesupport.cpp]: Also, I don't think this needs to be Nestmate specific.  If `checkVisibility` has a pending exception, we want to throw that one.

Comment (gacholio) [reply runtime/vm/resolvesupport.cpp]: If canRunJavaCode is false, then we'd better not be setting an exception in checkVisibility - hopefully resolveFlags takes care of that.

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk11

**Comment by DanHeidinga** [comment]
The JDK11 failures are the known set (really, really wish that set was 0!) Merging

**Pull request merged**